src/main/kotlin/org/taktik/icure/entities/base/HasDataAttachments.kt

```diff
@@ -0,0 +1,48 @@
+package org.taktik.icure.entities.base
+
+import org.taktik.icure.entities.Document
+import org.taktik.icure.entities.embed.DataAttachment
+import org.taktik.icure.entities.embed.DeletedAttachment
+import org.taktik.icure.entities.utils.MergeUtil
+
+/**
+ * Interface for entities which store part of their data as attachments.
+ */
+interface HasDataAttachments<T : HasDataAttachments<T>> : StoredDocument {
+	/**
+	 * All data attachments associated to this entity. The key in this map represents the key of the attachment.
+	 * By convention, in entities where there is a main attachment (e.g. [Document]) the key of the main attachment
+	 * should be equal to the entity id, in order to avoid accidental collisions.
+	 */
+	val dataAttachments: Map<String, DataAttachment>
+
+	/**
+	 * History of all deleted attachments.
+	 */
+	// TODO do we actually want the history in all entities with data attachments?
```

Comment:
If I'm correct, only Document is using attachments for now. If in the future, we have another kind of entities using them and for which we don't want to keep the history of deleted attachments, we'll create a second interface specific to deletedAttachments (Such as `HasDeletedAttachmentsHistory` for example) where we'll put all the logic related to deleted attachments